### ISSUE TITLE
[Azure.ResourceManager.AppConfiguration] Multiple Tests Failing Due to Core Change

### ISSUE THREAD

**Issue by jsquire** [Azure/azure-sdk-for-net#29438]
# Summary

This morning's AppConfiguration test run failed due to a large number of the management tests failing with the following:

## Error:
```text
System.ArgumentNullException : Value cannot be null.
Parameter name: location
```

## Stack Trace:
```text
 at Azure.Core.AzureLocation..ctor(String location) in D:\a\_work\1\s\sdk\core\Azure.Core\src\AzureLocation.cs:line 296
   at Azure.Core.AzureLocation.op_Implicit(String location) in D:\a\_work\1\s\sdk\core\Azure.Core\src\AzureLocation.cs:line 379
   at Azure.ResourceManager.AppConfiguration.Tests.ConfigurationStoreCollectionTests.<GetAllTest>d__8.MoveNext() in D:\a\_work\1\s\sdk\appconfiguration\Azure.ResourceManager.AppConfiguration\tests\Tests\ConfigurationStoreCollectionTests.cs:line 73
--- End of stack trace from previous location where exception was thrown ---
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
   at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
   at NUnit.Framework.Internal.TaskAwaitAdapter.GenericAdapter`1.GetResult()
   at NUnit.Framework.Internal.AsyncToSyncAdapter.Await(Func`1 invoke)
   at NUnit.Framework.Internal.Commands.TestMethodCommand.RunTestMethod(TestExecutionContext context)
   at NUnit.Framework.Internal.Commands.TestMethodCommand.Execute(TestExecutionContext context)
   at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.<>c__DisplayClass1_0.<Execute>b__0()
   at NUnit.Framework.Internal.Commands.BeforeAndAfterTestCommand.RunTestMethodInThreadAbortSafeZone(TestExecutionContext context, Action action)
```

Because the management and client libraries share the same test pipeline, this impacts the ability to monitor health of both.

# References and Related

- [Test Run - 20220623.1](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1662597&view=results)
- [PR that caused the issue](https://github.com/Azure/azure-sdk-for-net/pull/29434)

**Comment by jsquire** [comment]
//fyi: @ArthurMa1978, @m-nash

**Comment by Yao725** [comment]
This should be fixed by #29455, the tests of Azure.ResourceManager.AppConfiguration will be skiped in live mode so the test pipeline will be unblocked.
